Update README.md and obs script (#14)
ecmwf-lab/climetlab-s2s-ai-challenge: README.md

## Datasets description

There are four datasets provided for this challenge. As we are aiming at bringing together the two communities of Machine Learning and Weather Prediction, they have been aliases to use both two points of views:

| ML                          | NWP                          |                                                        |
| --------------------------- | ---------------------------- | ------------------------------------------------------ |
| `training-input`            | `hindcast-input`             | Training  dataset (input for training the ML models)   |
| `training-output-reference` | `hindcast-like-observations` | Training dataset (output for training the ML models)   |
| `training-output-benchmark` | `hindcast-benchmark`         | Benchmark output (on the training dataset) (NOT YET AVAILABLE)     |

> | `test-input`                | `forecast-input`             | Test dataset (DO NOT use for training)                              |

Reviewer: No. Do not use for training the ML model, but do not use to design them. Do no use to talk about performance of ML model. Do not use to have ideas about what to do. Do not use it during the challenge.

Author: Dont they need the test-input to create their ML-based predictions? I think they need and thats why it is called forecast-input.

  zarr (missing)
  
### Observations (Reference Output)
The `hindcast-like-observations` (`training-output-reference`) dataset.
The `forecast-like-observations` (`test-output-reference`) dataset.

The observations are the ground truth to compare with the ML model output and evaluate them. It consists in observation from instruments of [temperature](http://iridl.ldeo.columbia.edu/SOURCES/.NOAA/.NCEP/.CPC/.temperature/.daily/) and accumulated total [precipitation](http://iridl.ldeo.columbia.edu/SOURCES/.NOAA/.NCEP/.CPC/.UNIFIED_PRCP/.GAUGE_BASED/.GLOBAL/.v1p0/.extREALTIME/.rain/). The [NOAA CPC](https://www.cpc.ncep.noaa.gov/) datasets were downloaded from [IRIDL](iridl.ldeo.columbia.edu/). We provide observations in the same dimensions as the forecasts/hindcasts to have an easy match of forecasts/hindcast and ground truth. [See the script for technical details](https://github.com/ecmwf-lab/climetlab-s2s-ai-challenge/tree/main/tools/observations).

Generally speaking, only data available when the forecast is issued can be used by the ML models to perform their forecast:

> __Rule 1 : Observed data beyond the forecast date should not be used for prediction, for instance a forecast starting on 2020/07/01 should not use observed data beyond 2020/07/01).__

Author: maybe we could link here to the rules: https://s2s-ai-challenge.github.io/#rules

Reviewer: ok,. README updated.

> __Rule 1 : Observed data beyond the forecast date should not be used for prediction, for instance a forecast starting on 2020/07/01 should not use observed data beyond 2020/07/01).__

Author: this sounds like there are also other rules here? I think you had here more in the past, but I see this now as the only rule.

### Hindcast input (Training input)

The `hindcast-input`(`training-input`) dataset consists of data from three different models: ECMWF (ecmf), ECCC (cwao), NCEP (kwbc).
Use `origin="ecmwf"` (model name) or `origin="ecmwf"` (center name).
These data are hindcast data. This is used as the input for training the ML models.
This dataset is available as `grib`, `netcdf` or `zarr`.

> In this dataset, the data is available from 1999 for the oldest, to 2019/12/31 for the most recent. <!-- which model is available from 1998? -->

Author: so 1999 is fine.

  - ECMWF hindcast data
    - `forecast_time`: from 2000/01/02 to 2019/12/31, weekly every 7 days (every Thurday).
    - `lead_time`: 0 to 46 days
    - `valid_time` (`forecast_time` + `lead_time`): from 2000/01/02 to 2020/02/13
    - availables parameters : `2t/ci/gh/lsm/msl/q/rsn/sm100/sm20/sp/sst/st100/st20/t/tcc/tcw/tp/ttr/u/v`
  - ECCC hindcast data 
    - `forecast_time`: from to , weekly every 7 days (every Thurday).
    - `lead_time`: 1 to 32 days
    - `valid_time` (forecast_time + lead_time): from 
    - availables parameters: `2t/ci/gh/lsm/msl/q/rsn/sp/sst/t/tcc/tcw/tp/ttr/u/v`

>     - parameter not available: sm20, sm100, st20, st100

Author: ```suggestion
    - parameters not available: sm20, sm100, st20, st100
```

    - `lead_time` : 1 to 44 days
    - `valid_time` (`forecast_time` + `lead_time`): from 1999/01/07 to 2011/02/11
    - availables parameters: `2t/ci/gh/lsm/msl/q/sm100/sm20/sp/sst/st100/st20/t/tcc/tcw/tp/ttr/u/v`
    - parameter not available: `rsn`

 List of files :
  [grib](https://storage.ecmwf.europeanweather.cloud/s2s-ai-challenge/data/training-input/0.3.0/grib/index.html),
  [netcdf](https://storage.ecmwf.europeanweather.cloud/s2s-ai-challenge/data/training-input/0.3.0/netcdf/index.html),
  zarr (missing)

### Forecast input (Test input)
The `forecast-input` (`test-input`) dataset consists also in data from three different models: ECMWF (ecmf), ECCC (cwao), NCEP (eccc), for different dates.
These data are forecast data.
This could be used the input for applying the ML models in order to generate the output which is submitted for the challenge.
Using data from earlier date that 2020/01/01 is also allowed during the prediction phase.
The forecast start dates in this dataset are from 2020/01/02 to 2020/12/31.
  - For all 3 models: 
    - `forecast_time`: from 2020/01/02 to 2020/12/31, weekly every 7 days (every Thurday).

>     - `valid_time` (`forecast_time` + `lead_time`): from 2020/01/02 to 2021/02/xx <!-- valid_time extends into 2021 -->

Author: doesnt it depend on the model as they have different leads? therefore I wrote xx as a placeholder

The `hindcast-like-observations` (`training-output-reference`) dataset.
The `forecast-like-observations` (`test-output-reference`) dataset.

The observations are the ground truth to compare with the ML model output and evaluate them. It consists in observation from instruments of temperature and accumulated total precipitation. (TODO add more descriptions) (point to the scripts to create them ? TODO).

> Generally speaking, only past data can be used by the ML models to perform their forecast :

Author: they can also use the forecasts issued on the same date. see my suggestion below

During forecast phase (i.e. the evaluation phase using the forecast-input dataset), 2020 observation data is used. Rule 1 still stands : Observed data beyond the forecast start date should not be used for prediction.

### Forecast Benchmark (Benchmark output)

The `forecast-benchmark` (`test-output-benchmark`) dataset is a probabilistic re-calibrated ECMWF forecast with categories `below normal`, `near normal`, `above normal`.

The benchmark consists in applying to the `forecast-input` a simple re-calibration of from the mean of the hindcast (training) data.

The benchmark data is available as follows :
  - `forecast_time`: from 2020/01/02 to 2020/12/31, weekly every 7 days (every Thurday).
  - `lead_time`: 14 days and 28 days, where this day represents the first day of the biweekly aggregate
  - `valid_time` (`forecast_time` + `lead_time`): from 2020/01/01 to 2021/01/29

>   - `category`: `'below normal'`, `'near normal'`, `'above normal'` <!-- todo: implement in stoarge and/or climetlab @florian -->

Author: for now I hard code this in my example

>   - `category`: `'below normal'`, `'near normal'`, `'above normal'` <!-- todo: implement in stoarge and/or climetlab @florian -->

Author: this todo is not visible. its a comment

*For benchmark datasets (reference output)*, the pattern will be similar to https://storage.ecmwf.europeanweather.cloud/s2s-ai-challenge/data/training-output-benchmark/{parameter}-weeks-{weeks}.nc (NOT AVAILABLE YET)

- {datasetname} : In the URLs the dataset name must follow the ML naming (`training-input` or `training-output-reference` or `training-output-benchmark`).
- {format} is `netcdf`. Training output is also available as GRIB file,  using `format='grib'` and replacing `".nc"` by `".grib"`

> - {parameter} is `t2m` for surface temperature at 2m, "tp" for total precipitation using CF convention.

Author: ```suggestion
- {parameter} is `t2m` for [surface temperature at 2m](https://confluence.ecmwf.int/display/S2S/S2S+Surface+Air+Temperature), `tp` for [total precipitation](https://confluence.ecmwf.int/display/S2S/S2S+Total+Precipitation)
```

- {origin} : `ecmwf` or `eccc` or `ncep` <!-- we should have a clean table for this once -->

> - {fctype} : "hindcast" ("forecast" for test dataset). <!-- is this needed in the API? IMO not -->

Author: can we delete `fctype`? this is not needed anymore, at least in the external API? @floriankrb

> - {fctype} : "hindcast" ("forecast" for test dataset). <!-- is this needed in the API? IMO not -->

Author: ```suggestion
```
